cmd/batch-rotate.go

```diff
@@ -336,7 +336,7 @@ func (r *BatchJobKeyRotateV1) Start(ctx context.Context, api ObjectLayer, job Ba
 			return false
 		}
 		if r.Flags.Filter.KMSKeyID != "" {
-			if v, ok := info.Metadata[xhttp.AmzServerSideEncryptionKmsID]; ok && strings.TrimPrefix(v, crypto.ARNPrefix) != r.Flags.Filter.KMSKeyID {
+			if v, ok := info.Metadata[xhttp.AmzServerSideEncryptionKmsID]; ok && strings.TrimPrefix(v, crypto.ARNPrefix) == r.Flags.Filter.KMSKeyID {
```

Comment:
I do not think this is correct. It is correct to rename skip to select object, but here we are returning false when the kms key id matches?